zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -870,6 +939,90 @@ impl<F: Field> MPTConfig<F> {
+                    assignf!(region, (self.fixed_table[3], offset) => true.scalar())?;
+                    assignf!(region, (self.fixed_table[4], offset) => false.scalar())?;
+                    offset += 1;
+                }
+                // 248
+                // long list
+                assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                assignf!(region, (self.fixed_table[1], offset) => 248i32.scalar())?;
+                assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[3], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[4], offset) => false.scalar())?;
+                offset += 1;
+                // 249
+                // very long list
+                assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::RLP.scalar())?;
+                assignf!(region, (self.fixed_table[1], offset) => 249i32.scalar())?;
+                assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[3], offset) => false.scalar())?;
+                assignf!(region, (self.fixed_table[4], offset) => true.scalar())?;
+                offset += 1;
+
+                // KEY_PREFIX_ODD: u8 = 0b0001_0000;
+                // first byte of ext key >> 4 == 1
+                for ind in 0..256 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::ExtOddKey.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                    if 16 <= ind && ind < 32 {
+                        assignf!(region, (self.fixed_table[2], offset) => true.scalar())?;
+                    } else {
+                        assignf!(region, (self.fixed_table[2], offset) => false.scalar())?;
+                    }
+                    offset += 1;
+                }
+
+                for ind in 0..2 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::StartNode.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                }
+                for ind in 0..21 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::BranchNode.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                }
+                for ind in 0..12 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::AccountNode.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                }
+                for ind in 0..6 {
+                    assignf!(region, (self.fixed_table[0], offset) => FixedTableTag::StorageNode.scalar())?;
+                    assignf!(region, (self.fixed_table[1], offset) => ind.scalar())?;
+                }
```

Comment:
I'm not sure I actually fully understand the lookup use for this (seems easily replaceable by a hard-coded constant in the circuit for the `Count`) but this doesn't update `offset` so this will overwrite the same row over and over again.

Reply:
My bad, the `offset` is a mistake. In general since I took the `q_row` approach, I was trying to constrain `q_row` dynamically so that in each state it's in a designated range, but I wanted to avoid using a lot of columns to track differences in offsets. But I'll remove these since you clarify with the table that has `is_start`, `is_branch`, `is_account`, and `is_storage`